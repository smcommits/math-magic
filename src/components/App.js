/* eslint-disable no-unused-vars */
/* eslint-disable react/no-access-state-in-setstate */
import React from 'react';

import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate';
import * as StyledComponent from './componentStyles/styledComponent';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: '',
      next: '',
      operation: '',
    };

    this.handleClick = this.handleClick.bind(this);
  }

  componentDidMount() {
    document.addEventListener('keypress', (e) => {
      this.handleKeyPress(e.key);
    });
  }

  handleClick(buttonName) {
    this.setState(calculate(this.state, buttonName));
    console.log(this.state);
  }

  handleKeyPress(buttonName) {
    this.setState(calculate(this.state, buttonName));
  }

  render() {
    const { total, next, operation } = this.state;
    return (
      <StyledComponent.FlexContainer style={{ maxWidth: '700px' }}>
        <Display total={total} next={next} />
        <ButtonPanel clickHandler={this.handleClick} />
      </StyledComponent.FlexContainer>
    );
  }
}

export default App;
