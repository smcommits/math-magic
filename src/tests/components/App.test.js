import React from 'react';
import renderer from 'react-test-renderer';
import App from '../../components/App';

describe('App Componenet', () => {
  it('it always renders the same markup', () => {
    const componentTree = renderer
      .create(<App />)
      .toJSON();
    expect(componentTree).toMatchSnapshot();
  });
});
