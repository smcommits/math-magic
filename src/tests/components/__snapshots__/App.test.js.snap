// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Componenet it always renders the same markup 1`] = `
<div
  className="sc-bdnxRM iWCGnI"
  style={
    Object {
      "maxWidth": "700px",
    }
  }
>
  <div
    className="sc-bdnxRM iWCGnI"
  >
    <h4
      className="display"
    >
      
    </h4>
  </div>
  <div
    className="sc-bdnxRM iWCGnI"
  >
    <div>
      <button
        className="sc-eCApnc gsChNR"
        color="white"
        onClick={[Function]}
        type="submit"
      >
        AC
      </button>
      <button
        className="sc-jSFjdj eajors"
        color="white"
        onClick={[Function]}
        type="submit"
      >
        +/-
      </button>
      <button
        className="sc-gKAaRy jaNjmA"
        color="white"
        onClick={[Function]}
        type="submit"
      >
        %
      </button>
      <button
        className="sc-iCoGMd laCQWH"
        color="orange"
        onClick={[Function]}
        type="submit"
      >
        /
      </button>
    </div>
    <div>
      <button
        className="sc-fujyAs cRPXPR"
        color="white"
        onClick={[Function]}
        type="submit"
      >
        7
      </button>
      <button
        className="sc-pNWdM fQHWgf"
        color="white"
        onClick={[Function]}
        type="submit"
      >
        8
      </button>
      <button
        className="sc-jrsJWt kqmdox"
        color="white"
        onClick={[Function]}
        type="submit"
      >
        9
      </button>
      <button
        className="sc-kEqXSa djbGhZ"
        color="orange"
        onClick={[Function]}
        type="submit"
      >
        x
      </button>
    </div>
    <div>
      <button
        className="sc-iqAclL bLdGMp"
        color="white"
        onClick={[Function]}
        type="submit"
      >
        4
      </button>
      <button
        className="sc-crzoAE igWTtX"
        color="white"
        onClick={[Function]}
        type="submit"
      >
        5
      </button>
      <button
        className="sc-dIsUp cWQvyw"
        color="white"
        onClick={[Function]}
        type="submit"
      >
        6
      </button>
      <button
        className="sc-bqGGPW ezTzYg"
        color="orange"
        onClick={[Function]}
        type="submit"
      >
        -
      </button>
    </div>
    <div>
      <button
        className="sc-ksluID bjphtf"
        color="white"
        onClick={[Function]}
        type="submit"
      >
        1
      </button>
      <button
        className="sc-hBMUJo gZRyXw"
        color="white"
        onClick={[Function]}
        type="submit"
      >
        2
      </button>
      <button
        className="sc-fnVZcZ hZfcle"
        color="white"
        onClick={[Function]}
        type="submit"
      >
        3
      </button>
      <button
        className="sc-fFSPTT eWpmWj"
        color="orange"
        onClick={[Function]}
        type="submit"
      >
        +
      </button>
    </div>
    <div>
      <button
        className="sc-bkbkJK kKBwOE"
        color="white"
        onClick={[Function]}
        type="submit"
      >
        0
      </button>
      <button
        className="sc-iemWCZ gEJsec"
        color="white"
        onClick={[Function]}
        type="submit"
      >
        .
      </button>
      <button
        className="sc-dIvrsQ gXPjxL"
        color="orange"
        onClick={[Function]}
        type="submit"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
